packages/grafana-data/src/transformations/matchers/nameMatcher.ts

```diff
@@ -4,32 +4,66 @@ import { FieldMatcherInfo, FrameMatcherInfo } from '../../types/transformations'
 import { stringToJsRegex } from '../../text/string';
 import { getFieldDisplayName } from '../../field/fieldState';
 
-// General Field matcher
-const fieldNameMacher: FieldMatcherInfo<string> = {
+export interface FieldNameMatcherOptions {
+  pattern?: string;
+  names?: string[];
```

Comment:
This is sadly used in the FieldNameMatcherEditor as the persisted model, so this change breaks all overrides defined thus far and complicates the override model for the simple one field name match case that we use today. 

Feels like we should split up the matchers a bit, 1 for regex, one for a single name, 1 for multi-names. But question is, is there time for that? What should we do for stable & now?